### PR TITLE
Add dropout into grouped_gemm_scale_softmax_gemm

### PR DESCRIPTION
**Pull request by guangzlu**
Added random number generator philox.
Philox is used to generate random numbers on device, and then we can use these random numbers to do dropout.
Added dropout in to grouped_gemm_scale_softmax_gemm, which is used to do training in mha.

### CODE REVIEW

File: include/ck/tensor_operation/gpu/block/blockwise_softmax.hpp [hunk 1]
```diff
@@ -108,6 +108,41 @@ struct BlockwiseSoftmax
         });
     }
 
+    template <typename CThreadBuffer>
+    __host__ __device__ void ApplyDropout(CThreadBuffer& in_thread_buf,
+                                          ushort p_dropout_16bits,
+                                          ck::philox ph,
```
Comment (ltqin) [inline include/ck/tensor_operation/gpu/block/blockwise_softmax.hpp:114]: why is ApplyDropout a function of BlockwiseSoftmax?

Comment (guangzlu) [reply include/ck/tensor_operation/gpu/block/blockwise_softmax.hpp:114]: Because dropout is did on the result of softmax. Some of the parameters of softmax can be reused in dropout, such as MRepeat and KRepeat in this function. With these two parameter, we can calculate how many elements are on a thread, and calculate the offset of an single element.
